packages/recommend-core/src/getBatchRecommendations.ts

```diff
@@ -71,13 +75,18 @@ export async function getBatchRecommendations<TObject>({
       recommendations = mapByScoreToRecommendations<ProductRecord<TObject>>({
         maxRecommendations,
         hits: splitResult.map((res) => res.hits).flat(),
+        indexName,
+        queryID: splitResult.at(0)?.queryID,
       });
     } else {
       recommendations = mapToRecommendations<ProductRecord<TObject>>({
         maxRecommendations,
         hits: splitResult.map((res) => res.hits),
         nrOfObjs: keyPair.value,
-      });
+        queryIDs: splitResult.map((res) => res.queryID),
+      })
+        .map((hit) => addIndexName(hit, indexName))
+        .map((hit, idx) => addAbsolutePosition(hit, idx));
```

Comment:
Nitpick: I doubt this will have a significant performance impact based on the limited number of recommendations usually returned but you could remove one `map` iteration and transform your hits in sequence the first iteration.

Reply:
I also thought about this 🤔 but with `n=30` I thought readability took precedence over optimization WDYT ?

Reply:
Makes sense, I agree with your point.